R/teal_card.R

```diff
@@ -89,7 +89,12 @@ c.teal_card <- function(...) {
   dots <- list(...)
   structure(
     Reduce(
-      f = function(u, v) append(u, if (inherits(v, "teal_card") || inherits(v, "list")) v else list(v)),
+      f = function(u, v) {
+        if (!inherits(v, "teal_card")) v <- as.teal_card(v)
+        result <- append(u, v)
+        attributes(result) <- modifyList(attributes(u) %||% list(), attributes(v))
+        result
+      },
       x = dots[-1],
       init = unclass(dots[[1]]) # unclass to avoid infinite recursion
```

Comment:
2 considerations here:

- We cannot use `NextMethod` inside another function
    - it needs to be at the top-level the function it's calling
- We cannot pass empty list as `init` parameter
    - append will take the class of the non-empty object.